main/src/cgeo/geocaching/ui/CacheListAdapter.java

```diff
@@ -426,8 +430,8 @@ public View getView(final int position, final View rowView, final ViewGroup pare
         else {
             holder.text.setText(cache.getName());
         }
-        holder.text.setCompoundDrawablesWithIntrinsicBounds(MapUtils.getCacheMarker(res, cache, cacheListType), null, null, null);
-
+        holder.cacheListType = cacheListType;
+        updateViewHolder(holder, cache, res);
```

Comment:
Wouldn't it be enough to pass `cacheListType` to `updateViewHolder`? Does it need to be stored in the holder itself?